router/session.py

```diff
@@ -0,0 +1,103 @@
+    session_start_date, session_start_timestamp = get_date(parsed_start_time), get_timestamp(parsed_start_time)
+    current_date, current_timestamp = get_current_date(), get_current_timestamp()
+    repeated_session_start_datetime = datetime(current_date.year, current_date.month, current_date.day, parsed_start_time.hour, parsed_start_time.minute, parsed_start_time.second)
+    if session_start_date <= current_date:
+        if repeat_schedule is None:
+            return session_start_timestamp <= current_timestamp + session_start_buffer_time
+        else:
+            return repeated_session_start_datetime <= current_timestamp + session_start_buffer_time
+    return False
+
+def is_end_time_valid(end_time: str, repeat_schedule: str):
+    """
+    Checks if session end time is valid
+    - If end time is given:
+        - If session end date is greater than or equal to current date:
+            - If session is not repeating, checks if the current timestamp is less than or equal to session end timestamp
+            - If session is repeating, builds a timestamp using current date and session end time and checks that with current timestamp
+    - Else, always returns True
+    """
+    if end_time is not None:
+        parsed_end_time = datetime.strptime(end_time,'%Y-%m-%dT%H:%M:%SZ')
+        session_end_date, session_end_timestamp = get_date(parsed_end_time), get_timestamp(parsed_end_time)
+        current_date, current_timestamp = get_current_date(), get_current_timestamp()
+        repeated_session_end_datetime = datetime(current_date.year, current_date.month, current_date.day, parsed_end_time.hour, parsed_end_time.minute, parsed_end_time.second)
+        if session_end_date >= current_date:
+            if repeat_schedule is not None:
+                return current_timestamp <= repeated_session_end_datetime
+            return current_timestamp <= session_end_timestamp
+        return False
+    return True
+
+
+def is_repeat_schedule_valid(repeat_schedule: str):
+    """
+    Checks if the repeating schedule matches to the current day.
+    """
+    if repeat_schedule is not None:
+        if repeat_schedule["type"] == "weekly":
+            return get_current_datetime().weekday() in repeat_schedule["params"]
+    return True
+
+@router.post("/get-session-data")
+def get_session_data(session_id: str):
+    """
+    API to get details about a session if the session is active.
+    Otherwise, returns False to denote session is not active.
+    """
+    query_params = {'session_id':session_id}
+    response = requests.get(session_db_url, params=query_params)
+    session_data = response.json()
+    if response.status_code == 200:
+        if session_data["is_active"] and is_start_time_valid(session_data["start_time"], session_data["repeat_schedule"]) and is_end_time_valid(session_data["end_time"], session_data["repeat_schedule"]) and is_repeat_schedule_valid(session_data["repeat_schedule"]):
+            return session_data
```

Comment:
why are we not using pydantic models here? As we're returning a complex object in an API call, I think it would be best to perform implicit validations using pydantic models here. The API will have a `response_model=` field which will take `BaseModel` pydantic class.
I would highly recommend this as this saves us a lot of time and helps us define the schema as well